packages/nimble-components/src/select/index.ts

```diff
@@ -102,6 +407,33 @@ export class Select extends FoundationSelect implements ErrorPattern {
             );
         }
     }
+
+    private updateSelectedIndexFromFilteredSet(): void {
+        if (
+            !this.options[this.selectedIndex]
+            && !this.committedSelectedOption
+        ) {
+            return;
+        }
+
+        const selectedItem = this.options[this.selectedIndex] ?? this.committedSelectedOption!;
```

Comment:
What would you think about rewriting this as:

```suggestion
        const selectedItem = this.options[this.selectedIndex] ?? this.committedSelectedOption;
        
        if (!selectedItem) {
            return;
        }
```

I believe that would keep the same behavior, but it would allow you to get rid of the non-null assertion.